### ISSUE TITLE
Large XML File near 7mil lines --- TAG-Highlight not working (works in v8.1.9.2 64-bit)

### ISSUE THREAD

**Issue by uhliro26** [notepad-plus-plus/notepad-plus-plus#11389]
### Description of the Issue

Large XML File near 7mil lines --- TAG-Highlight not working (works in v8.1.9.2 64-bit)

### Steps to Reproduce the Issue

1. open attached big XML (171neu2utm-21.zip)
2. turn on Language/XML
3. 

### Expected Behavior

Highlighting as in  v8.1.9.2 64-bit

### Actual Behavior

TAG-Highlight not working
It takes minutes after changing to XML
As a sidecar effect, if "remember current session on next startup" is active
than n++ seams to hang minutes because of session.xml tells to open file as xml

### Debug Information

Notepad++ v8.3.3   (64-bit)
Build time : Mar 13 2022 - 17:20:02
Path : C:\npp\notepad++.exe
Command Line : 
Admin mode : OFF
Local Conf mode : ON
Cloud Config : OFF
OS Name : Windows 10 Enterprise (64-bit) 
OS Version : 2009
OS Build : 19042.1586
Current ANSI codepage : 1252
Plugins : mimeTools.dll NppConverter.dll NppExport.dll 

It takes minutes after changing to XML to become at least the folding but no highlighting appears

bad in v8.3.3 64-bit
![image](https://user-images.githubusercontent.com/48444235/158347857-d7ddd3b2-a12e-4fb2-bd8c-5c09c4dc8fbf.png)

iO in v8.1.9.2 64-bit
opening, folding and highlighting within seconds!
![image](https://user-images.githubusercontent.com/48444235/158347575-9fbafd79-a254-4ece-b865-b57e58422655.png)

File:
[171neu2utm-21.zip](https://github.com/notepad-plus-plus/notepad-plus-plus/files/8251839/171neu2utm-21.zip)

**Comment by xomx** [comment]
@uhliro26 
Your file is bigger than the current [200MB styling limit](https://github.com/notepad-plus-plus/notepad-plus-plus/blob/master/PowerEditor/src/ScintillaComponent/Buffer.h#L27).

**Comment by uhliro26** [comment]
Thanks!
But bad to know this from outside of the code. I found no news about that.

This is to me a regression looking at v8.1.9.2 especialy the speed because folding isnt supressed.
So I have to live without >3.3 or go with two Versions!

could/should this limit user definable?!

**Comment by donho** [comment]
Well, it will be considered - let's see if there are more request for that.

**Comment by uhliro26** [comment]
THX

beside that bad, **why** ist the same XML so so so much slower with 8.3.x than with v8.1.9.2 ???

**Comment by d0vgan** [comment]
> > could/should this limit user definable?!
> I think that it could be changed from the hardcoded value to a N++ setting

I think everything, that can be configurable, must be configurable. Hard-coded values make the user helpless.

**Comment by d0vgan** [comment]
> beside that bad, **why** ist the same XML so so so much slower with 8.3.x than with v8.1.9.2 ???

I've experimented with an XML file of 16 MB (I'd found this big XML file using Everything from voidtools, this particular XML file is "C:\Program Files (x86)\Windows Kits\10\References\10.0.19041.0\Windows.Foundation.UniversalApiContract\10.0.0.0\ru\Windows.Foundation.UniversalApiContract.xml").
The XML file itself is opened quickly for the first time. But! When I click the vertical scrollbar and change the position in the file, it takes a few seconds just to change the position in the file! And when I clicked the Function List (the embedded one, not the plugin), Notepad++ just stopped responding for a few minutes (yes, minutes!).
I restarted Notepad++. This time it took around 8 seconds to open the XML file and then clicking the vertical scrollbar was not so dramatically slow. But if I clicked the vertical scrollbar near the end of the file, it again took a few seconds to change the position. Does it mean that Scintilla parses the XML structure in the background thread and this operation is time-consuming?
Anyway, clicking Function List again made Notepad++ to not respond for a few minutes.
And we are talking about an XML file of 16 MB! It's not even 160 MB!

**Comment by d0vgan** [comment]
Here is how AkelPad deals with such a file:
- there is one option for a maximum number of characters in the document;
- and there is another option for a maximum number of folds (i.e. foldable nodes).

When either the first or the second limitation is reached, the folding (and FunctionList-like) ability is disabled.

![AkelPad_Coder](https://user-images.githubusercontent.com/2848457/158587471-19f0431a-844e-418d-ad85-8696cb376b64.png)

**Comment by donho** [comment]
> And we are talking about an XML file of 16 MB! It's not even 160 MB!

@d0vgan Do you always have the same problem with v8.4.6  ?

**Comment by d0vgan** [comment]
> > And we are talking about an XML file of 16 MB! It's not even 160 MB!
> 
> @d0vgan Do you always have the same problem with v8.4.6 ?

I've just followed the instructions from my previous reply (changing the position in a large XML file, restarting Notepad++ and finally openiong the Funcion List). The behavior of Notepad++ 8.4.6 is the same as described above.
This time I was using another large XML file for experiments, ".nuget\packages\netstandard.library\2.0.3\build\netstandard2.0\ref\netstandard.xml" with a size of 15 MB.
Notepad++ is still occupying one core of the processor trying to show the Funcion List (I am not sure how many minutes passed since I tried to open it for this XML file), the memory consuption of Notepad++ is 140 MB and continues to grow.
